modules/govuk_mysql/templates/usr/local/bin/xtrabackup_s3_restore

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+exec 1> >(/usr/bin/logger -s -t $(basename $0)) 2>&1
+
+set -e
+
+DATE=$(date +%s)
+BASE_DIR=/tmp/mysql/basedir/$DATE/
+INCR_DIR=/tmp/mysql/incremental/$DATE/
+
+echo "Starting MySQL environment sync"
+
+# Create directories to restore into
+for i in $BASE_DIR $INCR_DIR; do
+  mkdir -p $i
+done
+
+# Pull the latest backups from S3
+envdir <%= @env_sync_envdir %> gof3r get --endpoint s3-<%= @aws_region -%>.amazonaws.com -b <%= @s3_bucket_name %> -k latest/base.xbcrypt | xbcrypt --encrypt-algo=AES256 --decrypt --encrypt-key="<%= @encryption_key %>" | xbstream -x -C $BASE_DIR
+
+envdir <%= @env_sync_envdir %> gof3r get --endpoint s3-<%= @aws_region -%>.amazonaws.com -b <%= @s3_bucket_name %> -k latest/incremental.xbcrypt | xbcrypt --encrypt-algo=AES256 --decrypt --encrypt-key="<%= @encryption_key %>" | xbstream -x -C $INCR_DIR
+
+# Decompress the backups
+for i in $BASE_DIR $INCR_DIR; do
+  innobackupex --decompress $i
+done
+
+# Prepare the data and apply the logs
+innobackupex --apply-log --redo-only $BASE_DIR
+innobackupex --apply-log $BASE_DIR --incremental-dir=$INCR_DIR
+
+# Disable Puppet because it will interfere with the next steps if it runs during
+# this period
+test -f /var/lib/puppet/state/agent_disabled.lock || govuk_puppet --disable
+
+# Stop the service before we copy the data across
+service mysql stop
+
+# Move the current directory to a backup location
+echo "Backing up current dir"
+mv /var/lib/mysql /var/lib/mysql-bak
+
+# Recreate the datadir
+mkdir /var/lib/mysql
+
+# Copy the restored data
+innobackupex --copy-back $BASE_DIR
+
+# Update to correct permissions
+echo "Correcting permissions"
+chown -R mysql.mysql /var/lib/mysql
+
+# Start the service
+service mysql start
+
+# Clean up
+echo "Cleaning up"
+rm -rf /var/lib/mysql-bak $BASE_DIR $INCR_DIR
+
+# Re-enable Puppet and run it to ensure passwords are updated
+govuk_puppet --enable && govuk_puppet
```

Comment:
Should this always re-enable puppet? Even if it wasn't the reason the disable is in place?